src/components/RenderHabits.js

```diff
@@ -0,0 +1,66 @@
+import { useContext } from "react";
+import styled from "styled-components";
+
+import UserContext from "../context/UserContext";
+import arrayDays from "../assets/shared/objectDays";
+
+export default function RenderHabits({ deleteHabits }) {
+    const { habits } = useContext(UserContext);
+    return (
+        <>
+            {habits.map((value, index) => <Habits key={index}>
+                <p>{value.name}</p>
+                <Weekdays>
+                    {arrayDays.day.map((element, index) => <Days color={() => {
+                        for (let i = 0; i < value.days.length; i++) {
+                            if (element.num === value.days[i]) return "#CFCFCF";
+                        }
```

Comment:
**problem**: Here we have a big problem, there are several ways to do this and a for is the worst one, I advise you to think a little more.